tests/instancetype_test.go

```diff
@@ -660,15 +663,283 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
+					instancetype := instancetypev1alpha2.VirtualMachineInstancetype{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+							Kind:       "VirtualMachineInstancetype",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "VirtualMachineInstancetype",
+						},
+						Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+							CPU: instancetypev1alpha2.CPUInstancetype{
+								Guest: expectedCores,
+							},
+							Memory: instancetypev1alpha2.MemoryInstancetype{
+								Guest: expectedMemory,
+							},
+						},
+					}
+					instancetypeBytes, err := json.Marshal(instancetype)
+					Expect(err).ToNot(HaveOccurred())
+
+					preference := instancetypev1alpha2.VirtualMachinePreference{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+							Kind:       "VirtualMachinePreference",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "VirtualMachinePreference",
+						},
+						Spec: instancetypev1alpha2.VirtualMachinePreferenceSpec{
+							CPU: &instancetypev1alpha2.CPUPreferences{
+								PreferredCPUTopology: instancetypev1alpha2.PreferCores,
+							},
+						},
+					}
+					preferenceBytes, err := json.Marshal(preference)
+					Expect(err).ToNot(HaveOccurred())
+
+					return instancetypeBytes, preferenceBytes
+				}),
+				Entry("v1alpha2 VirtualMachineClusterInstancetype and VirtualMachineClusterPreference", func() ([]byte, []byte) {
+					instancetype := instancetypev1alpha2.VirtualMachineClusterInstancetype{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+							Kind:       "VirtualMachineClusterInstancetype",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "VirtualMachineClusterInstancetype",
+						},
+						Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+							CPU: instancetypev1alpha2.CPUInstancetype{
+								Guest: expectedCores,
+							},
+							Memory: instancetypev1alpha2.MemoryInstancetype{
+								Guest: expectedMemory,
+							},
+						},
+					}
+					instancetypeBytes, err := json.Marshal(instancetype)
+					Expect(err).ToNot(HaveOccurred())
+
+					preference := instancetypev1alpha2.VirtualMachineClusterPreference{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+							Kind:       "VirtualMachineClusterPreference",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "VirtualMachineClusterPreference",
+						},
+						Spec: instancetypev1alpha2.VirtualMachinePreferenceSpec{
+							CPU: &instancetypev1alpha2.CPUPreferences{
+								PreferredCPUTopology: instancetypev1alpha2.PreferCores,
+							},
+						},
+					}
+					preferenceBytes, err := json.Marshal(preference)
+					Expect(err).ToNot(HaveOccurred())
+
+					return instancetypeBytes, preferenceBytes
+				}),
+			)
+		})
```

Comment:
I wonder if a simple read/write test across version will be valuable here:
1. create v1alpha1 api object, read in v1alpha2 and v1beta1
2. create v1alpha2 api object, read in v1alpha1 and v1beta1
3. create v1beta2 api object, read in v1alpha1 and v1alpha2

I dug up an old file, now removed in kubernetes/kubernetes as an example https://github.com/yastij/kubernetes/blob/bfa12a099287727112675155d4d2a4d4857738e2/test/integration/events/events_test.go of such an e2e test

Reply:
https://github.com/kubevirt/kubevirt/pull/9575/commits/662ff686505094573006db4aedb5d2fa870d71b4